apps/system/js/bluetooth_transfer.js

```diff
@@ -644,15 +644,17 @@ var BluetoothTransfer = {
     var storage = navigator.getDeviceStorage(storageType);
     var getreq = storage.get(filePath);
```

Comment:
And according to the implementation in Gecko:

Promise::All implementation goes through each value in the array and calls ::Resolve to extract a promise object from it:
https://dxr.mozilla.org/mozilla-central/source/dom/promise/Promise.cpp#975

If the value *is* a Promise, it returns the same object:
https://dxr.mozilla.org/mozilla-central/source/dom/promise/Promise.cpp#729

If the value *isn't* a Promise, it returns a new promise resolved to that value:
https://dxr.mozilla.org/mozilla-central/source/dom/promise/Promise.cpp#760

It doesn't matter if the object has a .then method.